website/integrations: add opencloud integration (#22497)
goauthentik/authentik: website/integrations/chat-communication-collaboration/opencloud/index.md

---
title: Integrate with OpenCloud
sidebar_label: OpenCloud
support_level: community
---

## What is OpenCloud?

> OpenCloud is an open-source content collaboration platform for storing, syncing, and sharing files, built on the Infinite Scale (oCIS) architecture.
>
> -- https://opencloud.eu

## Preparation

The following placeholders are used in this guide:

- `opencloud.company` is the FQDN of the OpenCloud installation.
- `authentik.company` is the FQDN of the authentik installation.

> - `opencloud` is the slug of the authentik application.

Reviewer: We don't usually put the slug in the placeholders like this. We use `<application_slug>` wherever it needs to be.

Seems confusing to mention it in the preparation section because the user won't have the application slug until after they create the application in authentik.

Author: That's fair. I always have mixed feelings on the slugs because it could be whatever the user puts, but usually, they'll just name it the application. It is frustrating for the guides sometimes but honestly we should just keep doing what we've been doing because it's such a minor thing

I updated this for you

Reviewer: It does feel like there's a better solution that what we're currently doing. We'll discuss on slack

> This guide covers the [`opencloud-compose`](https://github.com/opencloud-eu/opencloud-compose) Docker deployment. OpenCloud authenticates only through OpenID Connect.

Reviewer: ```suggestion
This guide covers integrating authentik with the [`opencloud-compose`](https://github.com/opencloud-eu/opencloud-compose) Docker deployment. OpenCloud only supports authentication via OpenID Connect (OIDC).
```

Author: updated

            - Type **Authorization**, **Strict**: `https://opencloud.company/oidc-silent-redirect.html`
            - Type **Post Logout**, **Strict**: `https://opencloud.company/`
        - **Signing Key**: select any available key.
        - **Scopes**: `openid`, `profile`, `email`.
        - **Invalidation flow** (under **Flow settings**): `default-invalidation-flow` (**Default - Invalidation flow**). The default provider invalidation flow does not end the authentik session, which causes OpenCloud to immediately log back in after logout. If you want to use `default-provider-invalidation-flow` and keep your authentik session on logout, remove the Post Logout Redirect URI that you set above.
3. Click **Submit**.

## OpenCloud configuration

In the `opencloud-compose` project, enable the external IdP overlay in `COMPOSE_FILE`. This replaces OpenCloud's built-in IdP, so login goes through authentik only.

```bash
COMPOSE_FILE=docker-compose.yml:idm/external-idp.yml:custom/authentik-roles.yml
```

Set the OIDC values in `.env`:

```bash
OC_DOMAIN=opencloud.company
IDP_DOMAIN=authentik.company

> IDP_ISSUER_URL=https://authentik.company/application/o/<slug>/

Reviewer: ```suggestion
IDP_ISSUER_URL=https://authentik.company/application/o/<application_slug>/
```

Author: updated

OC_OIDC_CLIENT_ID=web
OC_OIDC_CLIENT_SCOPES=openid profile email
WEBFINGER_WEB_OIDC_CLIENT_ID=web
WEBFINGER_WEB_OIDC_CLIENT_SCOPES=openid profile email
```

:::info
`WEBFINGER_WEB_OIDC_CLIENT_ID` is required — the Web UI reads its client ID from WebFinger, and login will not start if it is empty.
:::

Create `custom/authentik-roles.yml` to assign every user the default role:

```yaml
---
services:
    opencloud:
        environment:
            PROXY_ROLE_ASSIGNMENT_DRIVER: "default"
            GRAPH_ASSIGN_DEFAULT_USER_ROLE: "true"
```

Recreate the stack:

```bash
docker compose up -d
```

> ## Verification

Reviewer: ```suggestion
## Configuration verification
```

Author: updated
